CMakeLists.txt

```diff
@@ -168,317 +227,382 @@ configure_file(cmake/nheko.h config/nheko.h)
-    src/UserSettingsPage.cpp
-    src/WelcomePage.cpp
-    src/main.cpp
-)
-
-# ExternalProject dependencies
-set(EXTERNAL_PROJECT_DEPS "")
+	# Dialogs
+	src/dialogs/CreateRoom.cpp
+	src/dialogs/ImageOverlay.cpp
+	src/dialogs/PreviewUploadOverlay.cpp
+	src/dialogs/InviteUsers.cpp
+	src/dialogs/JoinRoom.cpp
+	src/dialogs/MemberList.cpp
+	src/dialogs/LeaveRoom.cpp
+	src/dialogs/Logout.cpp
+	src/dialogs/UserProfile.cpp
+	src/dialogs/ReadReceipts.cpp
+	src/dialogs/ReCaptcha.cpp
+	src/dialogs/RoomSettings.cpp
+
+	# Emoji
+	src/emoji/Category.cpp
+	src/emoji/ItemDelegate.cpp
+	src/emoji/Panel.cpp
+	src/emoji/PickButton.cpp
+	src/emoji/Provider.cpp
+
+	# Timeline
+	src/timeline/TimelineViewManager.cpp
+	src/timeline/TimelineModel.cpp
+	src/timeline/DelegateChooser.cpp
+
+	# UI components
+	src/ui/Avatar.cpp
+	src/ui/Badge.cpp
+	src/ui/LoadingIndicator.cpp
+	src/ui/InfoMessage.cpp
+	src/ui/FlatButton.cpp
+	src/ui/FloatingButton.cpp
+	src/ui/Label.cpp
+	src/ui/OverlayModal.cpp
+	src/ui/SnackBar.cpp
+	src/ui/RaisedButton.cpp
+	src/ui/Ripple.cpp
+	src/ui/RippleOverlay.cpp
+	src/ui/OverlayWidget.cpp
+	src/ui/TextField.cpp
+	src/ui/TextLabel.cpp
+	src/ui/ToggleButton.cpp
+	src/ui/Theme.cpp
+	src/ui/ThemeManager.cpp
+
+	src/AvatarProvider.cpp
+	src/Cache.cpp
+	src/ChatPage.cpp
+	src/CommunitiesListItem.cpp
+	src/CommunitiesList.cpp
+	src/EventAccessors.cpp
+	src/InviteeItem.cpp
+	src/LoginPage.cpp
+	src/Logging.cpp
+	src/MainWindow.cpp
+	src/MatrixClient.cpp
+	src/MxcImageProvider.cpp
+	src/ColorImageProvider.cpp
+	src/QuickSwitcher.cpp
+	src/Olm.cpp
+	src/RegisterPage.cpp
+	src/RoomInfoListItem.cpp
+	src/RoomList.cpp
+	src/RunGuard.cpp
+	src/SideBarActions.cpp
+	src/Splitter.cpp
+	src/popups/SuggestionsPopup.cpp
+	src/popups/PopupItem.cpp
+	src/popups/ReplyPopup.cpp
+	src/popups/UserMentions.cpp
+	src/TextInputWidget.cpp
+	src/TopRoomBar.cpp
+	src/TrayIcon.cpp
+	src/Utils.cpp
+	src/UserInfoWidget.cpp
+	src/UserSettingsPage.cpp
+	src/WelcomePage.cpp
+	src/main.cpp
+	)
+
 
 include(FeatureSummary)
 
-set(Boost_USE_STATIC_LIBS OFF)
-set(Boost_USE_STATIC_RUNTIME OFF)
-set(Boost_USE_MULTITHREADED ON)
+if(USE_BUNDLED_BOOST)
+	hunter_add_package(Boost COMPONENTS iostreams system thread)
+endif()
 find_package(Boost 1.70 REQUIRED
-             COMPONENTS atomic
-                        chrono
-                        date_time
-                        iostreams
-                        random
-                        regex
-                        system
-                        thread)
+	COMPONENTS iostreams
+	system
+	thread)
+if(USE_BUNDLED_ZLIB)
+	hunter_add_package(ZLIB)
+endif()
 find_package(ZLIB REQUIRED)
+if(USE_BUNDLED_OPENSSL)
+	hunter_add_package(OpenSSL)
+endif()
 find_package(OpenSSL REQUIRED)
-find_package(MatrixClient 0.3.0 REQUIRED)
-find_package(Olm 2 REQUIRED)
+if(USE_BUNDLED_MTXCLIENT)
+	include(FetchContent)
+	set(BUILD_LIB_EXAMPLES OFF CACHE INTERNAL "")
+	set(BUILD_LIB_TESTS OFF CACHE INTERNAL "")
+	FetchContent_Declare(
+		MatrixClient
```

Comment:
Do these inline commit definitions conflict with the versions specified in deps/cmake?

Reply:
There is no deps/cmake anymore

Reply:
Oh, I actually forgot to delete that folder.... I'll do it in a follow up